packages/ember-data/lib/system/store.js

```diff
@@ -1045,18 +1045,11 @@ Store = Service.extend({
   */
   all: function(modelName) {
     var typeClass = this.modelFor(modelName);
-    var typeMap = this.typeMapFor(typeClass);
-    var findAllCache = typeMap.findAllCache;
 
-    if (findAllCache) {
-      this.recordArrayManager.updateFilter(findAllCache, typeClass);
-      return findAllCache;
-    }
-
-    var array = this.recordArrayManager.createRecordArray(typeClass);
+    var liveRecordArray = this.recordArrayManager.liveRecordArrayFor(typeClass);
+    this.recordArrayManager.updateFilter(liveRecordArray, typeClass);
```

Comment:
This call is needed so the initially created `RecordArray` is filled with the currently loaded records in the store.